core/json/src/main/scala/net/liftweb/json/Extraction.scala

```diff
@@ -265,11 +265,14 @@ object Extraction {
           val instance = jconstructor.newInstance(args.map(_.asInstanceOf[AnyRef]).toArray: _*)
           setFields(instance.asInstanceOf[AnyRef], json, jconstructor)
         } catch {
-          case e @ (_:IllegalArgumentException | _:InstantiationException) =>
-            fail("Parsed JSON values do not match with class constructor\nargs=" +
-                 args.mkString(",") + "\narg types=" + args.map(a => if (a != null)
-                   a.asInstanceOf[AnyRef].getClass.getName else "null").mkString(",") +
-                 "\nconstructor=" + jconstructor)
+          case exception: Exception =>
```

Comment:
Ah, you're right, so I should have a

```
case e =>
  throw e
```

For context, this is the result of a peculiarity about how "or" ing cases works. Without the outer match statement the type of the variable is `Throwable` even though they're both `Exception`.